elasticsearch/src/main/java/com/tcdi/zombodb/query_parser/QueryRewriter.java

```diff
@@ -379,33 +380,76 @@ private AggregationBuilder build(ASTTally agg) {
         }
     }
 
+    /**
+     * Determine if a particular field name is present in the index
+     *
+     * @param md index metadata
+     * @param fieldname field name to check for
+     * @return true if this field exists, false otherwise
+     */
+    private boolean hasDate(final IndexMetadata md, final String fieldname) {
+        return md.hasField(fieldname + DateSuffix);
+    }
+
     static class RangeSpecEntry {
         public String key;
         public Double from;
         public Double to;
     }
 
+    /**
+     * Container for date range aggregation spec
+     */
+    static class DateRangeSpecEntry {
+        public String key;
+        public String from;
+        public String to;
+    }
+
     private AggregationBuilder build(ASTRangeAggregate agg) {
+        final String fieldname = agg.getFieldname();
+        final IndexMetadata md = metadataManager.getMetadataForField(fieldname);
+
         try {
             ObjectMapper om = new ObjectMapper();
 
-            RangeSpecEntry[] rangeSpecEntries = om.readValue(agg.getRangeSpec(), RangeSpecEntry[].class);
-            RangeBuilder builder = new RangeBuilder(agg.getFieldname())
-                    .field(agg.getFieldname());
+            // if this is a date field, execute a date range aggregation
+            if (hasDate(md, fieldname)) {
+                final DateRangeBuilder dateRangeBuilder = new DateRangeBuilder(fieldname)
```

Comment:
Not really happy that this block is essentially repeated.  Might be a way to make it generic but the answer wasn't immediately obvious.